libvast/src/execution_node.cpp

```diff
@@ -316,38 +321,48 @@ struct exec_node_state : inbound_state_mixin<Input>,
     // Instantiate the operator with its input type.
     auto output_generator = op->instantiate(make_input_adapter(), *ctrl);
     if (not output_generator) {
-      return caf::make_error(
-        ec::unspecified, fmt::format("{} failed to instantiate operator: {}",
-                                     *self, output_generator.error()));
+      VAST_DEBUG("{} could not instantiate operator: {}", *self,
+                 output_generator.error());
+      return add_context(output_generator.error(),
+                         "{} failed to instantiate operator", *self);
     }
     if (not std::holds_alternative<generator<Output>>(*output_generator)) {
-      return caf::make_error(ec::logic_error,
-                             fmt::format("{} failed to instantiate operator",
-                                         *self));
+      return caf::make_error(
+        ec::logic_error, fmt::format("{} expected {}, but got {}", *self,
+                                     operator_type_name<Output>(),
+                                     operator_type_name(*output_generator)));
     }
     instance.emplace();
     instance->gen = std::get<generator<Output>>(std::move(*output_generator));
+    VAST_DEBUG("{} calls begin on instantiated operator", *self);
```

Comment:
I'm afraid this is too verbose. Let's remove it.

Reply:
I think it's helpful because `.begin()` often does non-trivial initialization. For example, `from file foo.json` tries to open the fail and calls `abort()` if this fails. I'll move this down to `VAST_TRACE` if that's alright.

Reply:
That's alright.